generators/server/templates/src/main/java/package/config/_MicroserviceSecurityConfiguration.java

```diff
@@ -120,9 +128,26 @@ public TokenStore tokenStore() {
       @Bean
       public JwtAccessTokenConverter jwtAccessTokenConverter() {
           JwtAccessTokenConverter converter = new JwtAccessTokenConverter();
-          converter.setSigningKey(jHipsterProperties.getSecurity().getAuthentication().getJwt().getSecret());
-
+          converter.setVerifierKey(getKeyFromAuthorizationServer());
           return converter;
       }
+
+      @Bean
+      public RestTemplate loadBalancedRestTemplate(RestTemplateCustomizer customizer) {
+          RestTemplate restTemplate = new RestTemplate();
+          customizer.customize(restTemplate);
+          return restTemplate;
+      }
+
+      @Inject
+      @Qualifier("loadBalancedRestTemplate")
+      private RestTemplate keyUriRestTemplate;
+
+      private String getKeyFromAuthorizationServer() {
+          HttpEntity<Void> request = new HttpEntity<Void>(new HttpHeaders());
+          return (String) this.keyUriRestTemplate
```

Comment:
is this done once or for every request? Currently i am keeping a "public.cert" in the resource servers resource path. So they have no need to communicate with the authserver to be able to verify the key. This is an security enhancement for me, as well as less request overhead